dali/operators/image/distortion/jpeg_compression_distortion_op_cpu.cc

```diff
@@ -68,17 +87,21 @@ void JpegCompressionDistortionCPU::RunImpl(workspace_t<CPUBackend> &ws) {
   thread_ctx_.resize(thread_pool.NumThreads());
 
   for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
-    thread_pool.AddWork(
-      [&, sample_idx, quality = quality_arg_[sample_idx].data[0]](int thread_id) {
-        auto &ctx = thread_ctx_[thread_id];
-        auto sh = in_shape.tensor_shape_span(sample_idx);
-        cv::Mat in_mat(sh[0], sh[1], CV_8UC3, const_cast<unsigned char*>(in_view[sample_idx].data));
-        cv::Mat out_mat(sh[0], sh[1], CV_8UC3, out_view[sample_idx].data);
-        cv::cvtColor(in_mat, out_mat, cv::COLOR_RGB2BGR);
-        cv::imencode(".jpg", out_mat, ctx.encoded, {cv::IMWRITE_JPEG_QUALITY, quality});
-        cv::imdecode(ctx.encoded, cv::IMREAD_COLOR, &out_mat);
-        cv::cvtColor(out_mat, out_mat, cv::COLOR_BGR2RGB);
-      }, in_shape.tensor_size(sample_idx));
+    auto shape = layout_shape(input.GetLayout(), in_shape.tensor_shape_span(sample_idx));
+    for (int elem_idx = 0; elem_idx < shape[0]; elem_idx++) {
+      int height = shape[1];
+      int width = shape[2];
+      int channels = shape[3];
+      size_t frame_size = volume(&shape[1], &shape[4]);
```

Comment:
I have doubts about the creating that 4-dim shape just to read shape[0] as number of frames.
A suggestion:
```
int ndim = shape.size();
int64_t nframes = ndim > 3 ? volume(&shape[0], &shape[ndim-3]) : 1;
int64_t frame_size = volume(&shape[ndim-3], &shape[ndim]);
int64_t height = shape[ndim-3];
int64_t width = shape[ndim-2];
int64_t channels = shape[ndim-1];
for (int elem_idx = 0; elem_idx < nframes; elem_idx++) {
  thread_pool.AddWork(...);
}
```

Reply:
Why would I use int64_t?

Reply:
@jantonguirao You don't need the conditional - `ndim-3` will work even without fframes - it will evaluate to 0 and you'll have an empty shape (which has a volume of 1 by definition).
Also, I'd rather use dimension indices from the layout:
```C++
int w_dim = layout.find('W');
int h_dim = layout.find('H');
int c_dim = layout.find('C');
int f_dim = layout.find('F');
int64_t nframes = volume(&shape[0], &shape[f_dim+1]);  // note that if f_dim is -1, this evaluates to an empty range, which has a volume of 1
int64_t frame_size = volume(&shape[f_dim+1], &shape[ndim]);
int64_t width = shape[w_dim];
int64_t height = shape[h_dim];
```

Reply:
@prak-nv out shapes are 64 bit, but I agree, for images it has little sense (I guess this would explode somewhere anyway)

Reply:
@prak-nv Just because it's the type used for shapes across DALI. You can use `auto`

Reply:
Thanks for explaining. Went with last version, though I dislike using npos like that for nframes - bit too smart.